src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -392,9 +404,10 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     if (!this.data) {
       return; // Wait for data
     }
-    this.rows = [...this.data];
     if (this.search.length > 0) {
       this.updateFilter();
+    } else {
+      this.rows = [...this.data];
```

Comment:
Alternatively we could define `rows` as an immutable.js `Immutable.List` to ensure that any change to the data triggers the change detection?

Reply:
I tough about that, but...

- My initial test failed.
- We only make changes to the rows in 2 places, filtering and adding 1 new row when we create something. And even the filtering would require recreating the List.
- There are no methods to update the List with a new array. We would have to create a new List each time we refresh the data.

Reply:
There are some other places that might work well with immutable.
So I might have another look at it in another PR.

Reply:
Ok. Anyway, I assume here we are always creating a new array/list, right?

Reply:
yes, we always create a new array when we get new data, so onPush should not cause problems.